edx_event_bus_kafka/internal/tests/test_consumer.py

```diff
@@ -88,46 +102,58 @@ def setUp(self):
+        mock_logger.error.assert_called_once_with("Kafka consumers not enabled, exiting.")
 
     @override_settings(
         EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL='http://localhost:12345',
         EVENT_BUS_KAFKA_BOOTSTRAP_SERVERS='localhost:54321',
         EVENT_BUS_TOPIC_PREFIX='prod',
     )
-    def test_consume_loop(self):
+    @patch('edx_event_bus_kafka.internal.consumer.logger', autospec=True)
+    def test_consume_loop(self, mock_logger):
         """
         Check the basic loop lifecycle.
         """
-        poll_call_count = 0
-
-        def fake_poll(*args, **kwargs):
-            nonlocal poll_call_count
-            poll_call_count += 1
-            # Return normally twice (to show looping), then throw (to
-            # show that we're not actually handling exceptions, at
-            # least at the moment.) If we start suppressing
-            # exceptions, we'll need some other way to break the loop
-            # for this test.
-            if poll_call_count >= 3:
+        emit_call_count = 0
+
+        def fake_emit(*args, **kwargs):
+            nonlocal emit_call_count, self
+            emit_call_count += 1
+            # Perform the following sequence of events on successive calls:
+            # 1) accept a message normally, 2) throw, 3) accept a message normally
+            # again, and 4) stop the loop so we can finish the unit test.
+            if emit_call_count == 1:
+                return
+            if emit_call_count == 2:
                 raise Exception("something broke")
-            return self.normal_message
+            if emit_call_count == 3:
+                return
+            else:
+                self.event_consumer._shut_down()  # pylint: disable=protected-access
 
-        with patch.object(self.event_consumer, 'process_single_message') as mock_process:
-            mock_consumer = Mock(**{'poll.side_effect': fake_poll}, autospec=True)
+        with patch.object(self.event_consumer, 'emit_signals_from_message', side_effect=fake_emit) as mock_emit:
+            mock_consumer = Mock(**{'poll.return_value': self.normal_message}, autospec=True)
             self.event_consumer.consumer = mock_consumer
-            with pytest.raises(Exception, match="something broke"):
-                self.event_consumer.consume_indefinitely()
+            self.event_consumer.consume_indefinitely()
 
         # Check that each of the mocked out methods got called as expected.
         mock_consumer.subscribe.assert_called_once_with(['prod-some-topic'])
-        assert mock_consumer.poll.call_args_list == [
-            call(timeout=1.0), call(timeout=1.0), call(timeout=1.0)
-        ]
-        assert mock_process.call_args_list == [
-            call(self.normal_message), call(self.normal_message)
-        ]
-        mock_consumer.close.assert_called_once_with()
+        assert mock_emit.call_args_list == [call(self.normal_message)] * 4
```

Comment:
Nit: Maybe use a variable to clarify the 4. I'm not even sure if the following is accurate:
```
num_of_expected_calls = 4
```
Did you notice that `num` is an abbreviation? So much for consistency. ;)

Reply:
Yeah, maybe `expected_loop_iterations = 4` or something.